core/src/main/java/org/wildfly/plugin/core/GalleonUtils.java

```diff
@@ -111,17 +111,43 @@ public static ProvisioningConfig buildConfig(Path provisioningFile) throws Provi
      * @param layers Layers to include.
      * @param excludedLayers Layers to exclude.
      * @param pluginOptions Galleon plugin options.
+     * @param log Log to log messages.
      * @return The provisioning config.
      * @throws ProvisioningException
      */
     public static ProvisioningConfig buildConfig(ProvisioningManager pm,
             List<FeaturePack> featurePacks,
             List<String> layers,
             List<String> excludedLayers,
-            Map<String, String> pluginOptions) throws ProvisioningException, IllegalArgumentException {
+            Map<String, String> pluginOptions, LogConsumer log) throws ProvisioningException, IllegalArgumentException {
         final ProvisioningConfig.Builder state = ProvisioningConfig.builder();
-        for (FeaturePack fp : featurePacks) {
+        boolean hasLayers = !layers.isEmpty();
+        boolean fpWithDefaults = true;
+        if (!hasLayers) {
+            // Check we have all feature-packs with default values only.
+            for (FeaturePack fp : featurePacks) {
+                if (fp.isInheritConfigs() != null ||
+                        fp.isInheritPackages() != null ||
+                        !fp.getIncludedConfigs().isEmpty() ||
+                        !fp.getExcludedConfigs().isEmpty() ||
+                        fp.isTransitive() ||
+                        !fp.getExcludedPackages().isEmpty() ||
+                        !fp.getIncludedPackages().isEmpty()) {
+                    fpWithDefaults = false;
+                    break;
+                }
+            }
+            if (fpWithDefaults) {
+                log.log(LogConsumer.Type.INFO, "No Layers have been provided and feature-packs don't override default "
```

Comment:
I understand what you mean. But for more fluent reading I would suggest something like:

```
No Layers are configured and feature-packs don't override default packages and configs inheritance. That mean the last feature-pack default configuration will be provisioned.

In case that is not what you would expect, you can configure inheritance of configurations in feature-packs using <inherit-configs>true|false</inherit-configs>
```

Reply:
Thank-you, I adjusted the wording.